usr/sbin/p11sak/p11sak.c

```diff
@@ -9495,12 +9503,25 @@ static CK_RV init_pkcs11(const struct p11sak_cmd *command)
     if (command == NULL || command->session_flags == 0)
         return CKR_OK;
 
-    if (pin == NULL)
-        pin = getenv(PKCS11_USER_PIN_ENV_NAME);
-    if (opt_force_pin_prompt || pin == NULL)
-        pin = pin_prompt(&buf_user_pin, "Please enter user PIN: ");
-    if (pin == NULL)
-        return CKR_FUNCTION_FAILED;
+    if (opt_no_login) {
+        if (opt_pin != NULL) {
+            warnx("Option '-p'/'--pin' is not allowed with '-N'/'--no-login'");
```

Comment:
OK, here we may have the case that the pin ist set via env variable, but let's just silently ignore it.